### PR TITLE
[MLIR][scf.parallel] Don't allow a tile size of 0

### PR DESCRIPTION
**Pull request by justinfargnoli**
Fix a crash reported in #64331. The crash is described in the following comment:

> It looks like the bug is being caused by the command line argument --scf-parallel-loop-tiling=parallel-loop-tile-sizes=0. More specifically, --scf-parallel-loop-tiling=parallel-loop-tile-sizes sets the tileSize variable to 0 on [this line](https://github.com/llvm/llvm-project/blob/7cc1bfaf371c4a816cf4e62fe31d8515bf8f6fbd/mlir/lib/Dialect/SCF/Transforms/ParallelLoopTiling.cpp#L67). tileSize is then used on [this line](https://github.com/llvm/llvm-project/blob/7cc1bfaf371c4a816cf4e62fe31d8515bf8f6fbd/mlir/lib/Dialect/SCF/Transforms/ParallelLoopTiling.cpp#L117) causing a divide by zero exception.

This PR will:

1. Call `signalPassFail()` when 0 is passed as a tile size.
2. Avoid the divide by zero that causes the crash. 

Note: This is my first PR for MLIR, so please liberally critique it.

### CODE REVIEW

File: mlir/lib/Dialect/SCF/Transforms/ParallelLoopTiling.cpp [hunk 1]
```diff
@@ -195,6 +195,9 @@ struct ParallelLoopTiling
   }
 
   void runOnOperation() override {
+    for (auto tileSize : tileSizes)
+      if (tileSize == 0)
+        signalPassFailure();
```
Comment (matthias-springer) [inline mlir/lib/Dialect/SCF/Transforms/ParallelLoopTiling.cpp:200]: I would emit an error message here (`mlir::emitError`).

**Comment by matthias-springer** [comment]
Can you also add a regression test case?

File: mlir/lib/Dialect/SCF/Transforms/ParallelLoopTiling.cpp (hunk 1 above)
Comment (justinfargnoli) [reply mlir/lib/Dialect/SCF/Transforms/ParallelLoopTiling.cpp]: I took a stab at this. Is there a way to specify that the location of the error message is from a command line argument? Or is the best way to go forward using `UnknownLoc`?

**Comment by justinfargnoli** [comment]
> Can you also add a regression test case?

I took a stab at this as well. Is there a way to use `-verify-diagnostics` to check for errors that are at an `UnkownLoc`? Or is `XFAIL: *` the best alternative?

**Comment by justinfargnoli** [comment]
Pinging for another round of review

File: mlir/test/Dialect/SCF/parallel-loop-tiling-check-tile-size.mlir [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// RUN: mlir-opt %s -pass-pipeline='builtin.module(func.func(scf-parallel-loop-tiling{parallel-loop-tile-sizes=0,0}))' -split-input-file -verify-diagnostics
+
+// XFAIL: *
```
Comment (matthias-springer) [inline mlir/test/Dialect/SCF/parallel-loop-tiling-check-tile-size.mlir:3]: You can check the expected error message with `// expected-error @below{{error message}}`

File: mlir/test/Dialect/SCF/parallel-loop-tiling-check-tile-size.mlir [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// RUN: mlir-opt %s -pass-pipeline='builtin.module(func.func(scf-parallel-loop-tiling{parallel-loop-tile-sizes=0,0}))' -split-input-file -verify-diagnostics
+
```
Comment (matthias-springer) [inline mlir/test/Dialect/SCF/parallel-loop-tiling-check-tile-size.mlir:2]: nit: Such files are usually called `parallel-loop-invalid.mlir`. Other error checking unit tests can then be added over time.

File: mlir/lib/Dialect/SCF/Transforms/ParallelLoopTiling.cpp (hunk 1 above)
Comment (matthias-springer) [reply mlir/lib/Dialect/SCF/Transforms/ParallelLoopTiling.cpp]: `UnknownLoc` sounds good here.

**Review by matthias-springer — APPROVED** [review]

File: mlir/test/Dialect/SCF/parallel-loop-tiling-check-tile-size.mlir (hunk 2 above)
Comment (justinfargnoli) [reply mlir/test/Dialect/SCF/parallel-loop-tiling-check-tile-size.mlir:3]: It looks like it isn't possible to use `expected-error` to check for an error at an `UnknownLoc`. If this is true, am I good to go forward with using `XFAIL`?

Comment (matthias-springer) [reply mlir/test/Dialect/SCF/parallel-loop-tiling-check-tile-size.mlir:3]: I didn't know that. Sounds good!

**Pull request merged**